ast/errors.go

```diff
@@ -0,0 +1,121 @@
+package ast
+
+import (
+	"strconv"
+	"strings"
+)
+
+// GetErrors will collect all of the errors from the construction of the AST.
+// If there were no errors, this will return nil. If the limit is set to
+// positive number, it will limit the number of errors to the given number. If
+// it is set to a negative number, it will return the last N errors. A value of
+// zero will return all errors.
+func GetErrors(root Node, limit int) []error {
```

Comment:
What is the motivation for the limit logic being the last errors? Why do we need the limit logic at all? An AST is expected to fit in RAM, since its error are a subset seems reasonable to always return all errors.

Reply:
This is more for a command line tool. It is pretty common for a compiler to limit itself to the first 10 errors. The  limit  logic is for the first errors and a negative number gets you the last errors, but I could just remove that.

Reply:
OK, yeah, when such a CLI tool exists that tool can limit the errors it displays. Lets remove it from here.

Reply:
Removed this. We can always add it back later.